diff --git a/app.js b/app.js
index 964bbe7..d2e0c65 100644
--- a/app.js
+++ b/app.js
@@ -50,20 +50,23 @@ app.configure('production', function(){
 var routes = require('./routes');
 app.get('/', routes.index);
 app.get('/about', routes.about);
-// app.get('/login', routes.login);
-// app.get('/logout', routes.logout);
 app.get('/sandbox', routes.sandbox);
 
-var authRoutes = require('./routes/auth');
-authRoutes.getPaths().forEach(function (path) {
-  app.get('/auth/' + path, authRoutes[path]);
-});
+app.get('/login', routes.login);
+app.get('/logout', routes.logout);
+app.get('/authenticate', routes.authenticate);
+app.get('/authcallback', routes.authcallback);
 
 var apiRoutes = require('./routes/api');
 apiRoutes.getPaths().forEach(function (path) {
   app.get('/api/' + path, apiRoutes[path]);
 });
 
+// var postRoutes = require('./routes/post');
+// postRoutes.getPaths().forEach(function (path) {
+//   app.get('/post/' + path, postRoutes[path]);
+// });
+
 
 
 
diff --git a/public/app/controller/frames/container/detail.js b/public/app/controller/frames/container/detail.js
index 3d1660e..9d85fef 100644
--- a/public/app/controller/frames/container/detail.js
+++ b/public/app/controller/frames/container/detail.js
@@ -323,6 +323,7 @@ app.controller.Detail.prototype.performActionInternal = function (e) {
 
 
 app.controller.Detail.MenuRecords_ = [
+  {content: 'E', key: 'watch', desc: 'このアイテムをウォッチリストに追加します'},
   {content: 'o', key: 'goToAuction', desc: 'オークションの商品ページを表示します'},
   {content: 'A', key: 'showPicture', desc: '写真を表示します'},
   {content: 'K', key: 'showDesc', desc: '商品説明を表示します'},
@@ -346,9 +347,12 @@ app.controller.Detail.prototype.handleMenuDismiss_ = function (e) {
 
 app.controller.Detail.prototype.handleMenuSelect_ = function (e) {
   switch(e.record.key) {
+    case 'watch':
+      // TODO
+      break;
     case 'goToAuction': 
       if (this.safeItemLink_) goog.window.open(this.safeItemLink_);
-      break; // TODO:
+      break;
     case 'showPicture': this.jumpToElement(this.imagesElementRef_); break;
     case 'showDesc': this.jumpToElement(this.descriptionElementRef_); break;
     case 'showDetail': this.jumpToElement(this.tableElementRef_); break;
diff --git a/public/app/model/model.js b/public/app/model/model.js
index af2ae19..df793dd 100644
--- a/public/app/model/model.js
+++ b/public/app/model/model.js
@@ -297,3 +297,9 @@ app.model.getAlignmentStyle = function (tabId) {
   return !!storage.get(app.model.KeyPrefix.TAB_ + tabId + 
               app.model.KeyPrefix._ISGRID_);
 };
+
+
+
+app.model.watchItem = function (item) {
+  app.model.Xhr.getInstance().post();
+};
diff --git a/routes/api.js b/routes/api.js
index 8fb13bd..e83954a 100644
--- a/routes/api.js
+++ b/routes/api.js
@@ -11,8 +11,6 @@ var YAHOO_ENDPOINTS = [
   '/AuctionWebService/V2/json/categoryLeaf',
   '/AuctionWebService/V2/json/auctionItem',
   '/AuctionWebService/V2/json/search',
-
-  '/AuctionWebService/V1/watchList'
 ];
 
 var MY_PATHS = [
diff --git a/routes/auth.js b/routes/auth.js
deleted file mode 100644
index d7148ea..0000000
--- a/routes/auth.js
+++ /dev/null
@@ -1,39 +0,0 @@
-
-var oa = require('../sources/core/oauth.js');
-
-exports.auth = function(req, res){
-  oa.getOAuthRequestToken(function(error, oauth_token, oauth_token_secret, results){
-    if (error) {
-      res.send("didn't work.<br />" + JSON.stringify(error))
-    } else {
-      req.session.oauth = {};
-      req.session.oauth.token = oauth_token;
-      req.session.oauth.token_secret = oauth_token_secret;
-      res.redirect('https://auth.login.yahoo.co.jp/oauth/v2/request_auth?oauth_token='+oauth_token)
-    }
-  });
-};
-
-exports.callback = function(req, res, next){
-  if (req.session.oauth) {
-    req.session.oauth.verifier = req.query.oauth_verifier;
-    var oauth = req.session.oauth;
-    oa.getOAuthAccessToken(oauth.token, oauth.token_secret, oauth.verifier, 
-        function(error, oauth_access_token, oauth_access_token_secret, results){
-          if (error){
-            res.send("yeah something broke.");
-          } else {
-            req.session.oauth.access_token = oauth_access_token;
-            req.session.oauth.access_token_secret = oauth_access_token_secret;
-            req.session.yahoo = results;
-            res.redirect("/");
-          }
-        }
-    );
-  } else
-    next(new Error("you're not supposed to be here."));
-};
-
-var paths = require('underscore').keys(module.exports);
-module.exports.getPaths = function () {return paths};
-
diff --git a/routes/index.js b/routes/index.js
index 4b22ba1..e249885 100644
--- a/routes/index.js
+++ b/routes/index.js
@@ -19,6 +19,10 @@ exports.index = function(req, res){
   // }
 };
 
+exports.sandbox = function(req, res){
+  res.render('sandbox');
+};
+
 exports.about = function (req, res) {
   res.render('about', {
     layout: !req.query.noLayout
@@ -37,10 +41,57 @@ exports.logout = function (req, res) {
   res.render('logout');
 };
 
-exports.sandbox = function(req, res){
-  res.render('sandbox');
+
+
+
+exports.authenticate = function(req, res){
+  oa.getOAuthRequestToken(function(error, oauth_token, oauth_token_secret, results){
+    if (error) {
+      res.send("didn't work.<br />" + JSON.stringify(error))
+    } else {
+      req.session.oauth = {};
+      req.session.oauth.token = oauth_token;
+      req.session.oauth.token_secret = oauth_token_secret;
+      res.redirect('https://auth.login.yahoo.co.jp/oauth/v2/request_auth?oauth_token='+oauth_token)
+    }
+  });
 };
 
+exports.authcallback = function(req, res, next){
+  if (req.session.oauth) {
+    req.session.oauth.verifier = req.query.oauth_verifier;
+    var oauth = req.session.oauth;
+    oa.getOAuthAccessToken(oauth.token, oauth.token_secret, oauth.verifier, 
+        function(error, oauth_access_token, oauth_access_token_secret, results){
+          if (error){
+            res.send("yeah something broke.");
+          } else {
+            req.session.oauth.access_token = oauth_access_token;
+            req.session.oauth.access_token_secret = oauth_access_token_secret;
+            req.session.yahoo = results;
+            res.redirect("/");
+          }
+        }
+    );
+  } else
+    next(new Error("you're not supposed to be here."));
+};
+
+// var paths = require('underscore').keys(module.exports);
+// module.exports.getPaths = function () {return paths};
+
+
+
+
+
+
+
+
+
+
+
+
+
 // exports.post = function (req, res) {
 //   if(req.session.oauth && req.session.oauth.access_token) {
 //     var text = req.body.text;
diff --git a/sources/core/oauth.js b/sources/core/oauth.js
index 963bbba..fc4b48f 100644
--- a/sources/core/oauth.js
+++ b/sources/core/oauth.js
@@ -7,7 +7,7 @@ module.exports = new OAuth(
     AUC_PRO.CONSUMER_KEY,
     AUC_PRO.CONSUMER_SECRET,
     "1.0",
-    "http://auc.stakam.net/auth/callback",
+    "http://" + AUC_PRO.DOMAIN + "/oauthcallback",
     "HMAC-SHA1");
 
 
diff --git a/views/index.jade b/views/index.jade
index e123c09..cd50842 100644
--- a/views/index.jade
+++ b/views/index.jade
@@ -5,10 +5,11 @@ div.toolbar
       a(href='/')
         img(src="/images/logo.png", title="Yan-Yan-Yafuokuについて")
 
-  // div.right
-  //   a(href='javascript:void(0)',title='watch list',tabindex='0').watch-tab.i E
-  //   div.logout 
-  //     a(href='/logout') logout
+  div.right
+    span(style="color:white") dev
+    // a(href='javascript:void(0)',title='watch list',tabindex='0').watch-tab.i E
+    // div.logout 
+    //   a(href='/logout') logout
 
   div.tabs
     div.tabs-content
diff --git a/views/login.jade b/views/login.jade
index 442afe9..f487424 100644
--- a/views/login.jade
+++ b/views/login.jade
@@ -3,4 +3,4 @@ div.login-wrap.container
     div.span8.offset3
       h1 Welcome to auc-pro.
     div.span6.offset3
-      a.btn.btn-primary.btn-large.btn-block(href='/auth/auth') login with yahoo! account
+      a.btn.btn-primary.btn-large.btn-block(href='/authenticate') login with yahoo! account
